archive/imo/imo1969_q1.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2020 Kevin Lacker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Lacker
+-/
+
+import tactic.linarith
+import tactic.norm_cast
+import tactic.ring
+
+open int
+open nat
+
+/-!
+# IMO 1969 Q1
+
+Prove that there are infinitely many natural numbers $a$ with the following property:
+the number $z = n^4 + a$ is not prime for any natural number $n$.
+
+The key to the solution is that you can factor z into the product of two polynomials,
+if a = 4*m^4.
+-/
+
+lemma factorization (m n : ℤ) : (n^2 + 2*m^2 - 2*n*m) * (n^2 + 2*m^2 + 2*n*m) = n^4 + 4*m^4
+:= by ring
```

Comment:
```suggestion
lemma factorization (m n : ℤ) :
  (n^2 + 2*m^2 - 2*n*m) * (n^2 + 2*m^2 + 2*n*m) = n^4 + 4*m^4 :=
by ring
```
The style guide wants `:=` before instead of after line breaks (-;
So if it doesn't fit on one line, then let's use 3 :grinning:

Reply:
Ah ok - I'll line break at the `:` then. Should be good now